src/Lachain.Core/RPC/HTTP/Web3/TransactionServiceWeb3.cs

```diff
@@ -190,12 +192,13 @@ public string SendRawTransaction(string rawTx)
             try
             {
                 var transaction = MakeTransaction(ethTx);
-                //TransactionUtils.SetChainId(41);
-                if (!ethTx.ChainId.SequenceEqual(new byte[] {(byte)(TransactionUtils.ChainId)}))
+                bool useNewChainId =
+                    HardforkHeights.IsHardfork_8Active(_stateManager.LastApprovedSnapshot.Blocks.GetTotalBlockHeight());
```

Comment:
If this tx is later added to the pool, then we should use GetTotalBlockHeight() + 1

Reply:
fixed